microshift_running_apps/microshift-greenboot.adoc

```diff
@@ -0,0 +1,89 @@
+:_content-type: ASSEMBLY
+[id="microshift-greenboot"]
+= The greenboot health check
+include::_attributes/attributes-microshift.adoc[]
+:context: microshift-greenboot
+
+toc::[]
+
+You might be challenged troubleshooting edge-device problems after software or an operating system upgrade fails. Edge devices are often in locations where direct serviceability is either limited or non-existent. In these cases, you can use greenboot to troubleshoot software problems for these devices. Greenboot is the Generic Health Check Framework for the `systemd` service on RPM-OSTree-based systems.
```

Comment:
Strictly speaking, greenboot does not troubleshoot device health. Instead, it's responsible for checking the device health and rolling back the system in case of a failed system or s/w upgrade.